libs/application-generic/src/services/analytic-logs/workflow-run/workflow-run.repository.ts

```diff
@@ -453,4 +453,247 @@ export class WorkflowRunRepository extends LogRepository<typeof workflowRunSchem
 
     return template.name.toLowerCase().replace(/\s+/g, '_');
   }
+
+  async getWorkflowVolumeData(
+    environmentId: string,
+    organizationId: string,
+    startDate: Date,
+    endDate: Date
+  ): Promise<Array<{ workflow_name: string; count: string }>> {
+    const query = `
+      SELECT 
+        workflow_name,
+        count(*) as count
+      FROM workflow_runs FINAL
+      WHERE 
+        environment_id = {environmentId:String} 
+        AND organization_id = {organizationId:String}
+        AND created_at >= {startDate:DateTime64(3)}
+        AND created_at <= {endDate:DateTime64(3)}
+      GROUP BY workflow_name
+      ORDER BY count DESC
+      LIMIT 5
+    `;
+
+    const params = {
+      environmentId,
+      organizationId,
+      startDate: LogRepository.formatDateTime64(startDate),
+      endDate: LogRepository.formatDateTime64(endDate),
+    };
+
+    const result = await this.clickhouseService.query<{
+      workflow_name: string;
+      count: string;
+    }>({
+      query,
+      params,
+    });
+
+    return result.data;
+  }
+
+  async getActiveSubscribersData(
+    environmentId: string,
+    organizationId: string,
+    startDate: Date,
+    endDate: Date,
+    previousStartDate: Date,
+    previousEndDate: Date
+  ): Promise<{ currentPeriod: number; previousPeriod: number }> {
+    // Query for current period
+    const currentPeriodQuery = `
+      SELECT count(DISTINCT external_subscriber_id) as count
+      FROM workflow_runs FINAL
+      WHERE 
+        environment_id = {environmentId:String} 
+        AND organization_id = {organizationId:String}
+        AND created_at >= {startDate:DateTime64(3)}
+        AND created_at <= {endDate:DateTime64(3)}
+    `;
+
+    // Query for previous period
+    const previousPeriodQuery = `
+      SELECT count(DISTINCT external_subscriber_id) as count
+      FROM workflow_runs FINAL
```

Comment:
non blocker: Instead of having a modular method, i think we could use the method twice instead.
once for the actual dates, then the second time for the previous dates.